static/src/javascripts/lib/defer-to-analytics.js

```diff
@@ -1,20 +1,19 @@
 // @flow
-import mediator from 'lib/mediator';
+import config from 'lib/config';
 
-let analyticsReady = false;
-
-mediator.on('analytics:ready', () => {
-    analyticsReady = true;
+const analyticsReady = new Promise(resolve => {
+    const check = () => {
+        if (config.get('modules.media.analyticsReady')) {
+            resolve(true);
+        } else {
+            setTimeout(check, 100);
```

Comment:
rather than polling, could you use a promise/queue mechanism as with this? https://github.com/guardian/frontend/blob/2172a416e196167fe8c864758f00ad7228274118/static/src/javascripts/lib/load-css-promise.js?

i'm very wary about adding a poll with no escape, esp not a pattern we've picked to use, but hack to fix one piece of code